core/src/subgraph/runner.rs

```diff
@@ -495,6 +496,17 @@ where
             return Err(BlockProcessingError::Deterministic(first_error.unwrap()));
         }
 
+        // For subgraphs with nonFatalErrors feature enabled,
+        // We need to update the nonFatalErrors field in the subgraph_deployment table.
+        if has_errors && is_non_fatal_errors_active {
+            let errors = deterministic_errors.clone();
+            self.inputs
+                .store
+                .update_non_fatal_errors(Some(errors.to_vec()))
+                .await
+                .context("Failed to update nonFatalErrors")?;
+        }
```

Comment:
This is going to cause trouble if `graph-node` gets stopped in between calling `transact_block_operations` and this logic here. All changes for a block need to be made through the call to `transact_block_operations`, and they need to happen in one database txn to make sure the db state is always consistent (either all changes for a block get written or not written)

Reply:
what would you suggest here? I could add a bool parameter `non_fatal_errors_enabled`  to `transact_block_operations` and then run it inside there if that sounds good